### PR TITLE
Document how pack interacts with http proxies

### PR DESCRIPTION
**Pull request by AidanDelaney**
Show how to both build an application behind a proxy and how an
application can be configured to access web APIs through a proxy.

### CODE REVIEW

**Comment by AidanDelaney** [comment]
Some buildpack operations [require the `http_` env vars](https://github.com/buildpacks/pack/blob/437c3f15a7501d432be94ebeabfb119a396ca549/build.go#L599) to be set.  This is independent of setting `~/.docker/config.json`.

I had expected that it is sufficient to set `httpProxy` in `~/.docker/config.json`.  However, when running `pack build --verbose sample-app --path samples/apps/java-maven --builder cnbs/sample-builder:bionic` the buildpack fails to install the JDK unless the `http_*` vars are also set.

File: content/docs/app-developer-guide/using-http-proxy.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+```console
+$ pack build sample-app --path samples/apps/java-maven --builder cnbs/sample-builder:bionic
+ERROR: failed to build: failed to fetch builder image 'index.docker.io/cnbs/sample-builder:bionic'
+: Error response from daemon: Get "https//registry-1.docker.io/v2/": context deadline exceeded
+```
+
+The `pack` tool uses the Docker daemon to manage the local image registry on your machine.  The `pack` tool will ask the Docker daemon to download buildpacks and images for you.  Because of this relationship, between `pack` and the Docker daemon, we need to configure the Docker daemon to use a HTTP proxy.
+
+The Docker project documents [how to configure configure the HTTP/HTTPS proxy](https://docs.docker.com/config/daemon/systemd/#httphttps-proxy) settings for the Docker daemon.  You should configure the `HTTP_PROXY` and `HTTPS_PROXY` environment variables as part of the Docker daemon startup.
```
Comment (jromero) [inline content/docs/app-developer-guide/using-http-proxy.md:26]: The link is linux specific which I find confusing as a Docker Desktop user. Based on [this link](https://docs.docker.com/desktop/mac/#resources), there are 2 ways to configure it for Docker Desktop. 

1. Setting [a proxy at the OS level](https://support.apple.com/guide/mac-help/enter-proxy-server-settings-on-mac-mchlp2591/mac) would be also propagate to the daemon on on Docker Desktop.
    > Docker Desktop detects HTTP/HTTPS Proxy Settings from macOS and automatically propagates these to Docker. 
2. Setting the Proxy settings in the [Resources](https://docs.docker.com/desktop/mac/#resources) section of Preferences.

Maybe it's worth denoting the difference based on the user setup. Docker Desktop and Linux.

**Review by jromero — COMMENTED** [review]
I really like the overall structure and information presented. Left a slight comment on what I think could be clarified. Let me know what you think.

File: content/docs/app-developer-guide/using-http-proxy.md (hunk 1 above)
Comment (AidanDelaney) [reply content/docs/app-developer-guide/using-http-proxy.md:26]: Thank you.  Yes, I will differentiate between Linux and Docker Desktop.

Comment (AidanDelaney) [reply content/docs/app-developer-guide/using-http-proxy.md]: In "Making `pack` proxy aware" I've provided Docker Desktop and Linux instructions.  I thought that was a better distinction than between Docker Desktop and Docker CE as some developers may be unfamiliar with the distinction between Desktop and CE.  I used a link to `dummies.com` for the Windows proxy configuration as I could not find an end-user focused guide in the official MS support documentation.

**Review by samj1912 — APPROVED** [review]

**Review by jromero — APPROVED** [review]

**Pull request merged**